plugins/org.jkiss.dbeaver.ext.databricks/src/org/jkiss/dbeaver/ext/databricks/DatabricksDataSource.java

```diff
@@ -35,4 +37,18 @@ public DatabricksDataSource(DBRProgressMonitor monitor,
     }
 
 
+    @Override
+    protected String getConnectionURL(DBPConnectionConfiguration connectionInfo) {
+        String url = super.getConnectionURL(connectionInfo);
+        if (!isLegacyDriver() && url.startsWith(DatabricksConstants.JDBC_LEGACY_URL_SUBPROTOCOL)) {
```

Comment:
```suggestion
        if (isLegacyDriver() && url.startsWith(DatabricksConstants.JDBC_LEGACY_URL_SUBPROTOCOL)) {
```

Reply:
Why then it is not isNativeDriver()?

private boolean isNativeDriver() {
        return CommonUtils.equalObjects("com.databricks.client.jdbc.Driver", getContainer().getDriver().getDriverClassName());
    }

Reply:
what if there will be a new databricks driver using the same url as in native one

Reply:
native is not legacy driver (i mean `databricks-native`)